### PR TITLE
Update user service.

### LINKED ISSUE #79: Update User creation (add URI)
Add user URI when a user is created due to new URI generator #45

### PR DESCRIPTION
**Pull request by niio972**
Create uri associated to created user. (fixes #79)

### CODE REVIEW

File: phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java [hunk 1]
```diff
@@ -564,7 +606,8 @@ private POSTResultsReturn checkAndInsertUserList(List<UserDTO> newUsers) throws
     }
 
     @Override
-    public POSTResultsReturn checkAndInsertList(List<UserDTO> newObjects) {
+    public POSTResultsReturn checkAndInsertList(List<UserDTO> newObjects
+    ) {
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java]: remove the enter line (indent)

File: phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java [hunk 2]
```diff
@@ -576,12 +619,14 @@ public POSTResultsReturn checkAndInsertList(List<UserDTO> newObjects) {
     }
 
     @Override
-    public POSTResultsReturn checkAndInsert(UserDTO newObject) {
+    public POSTResultsReturn checkAndInsert(UserDTO newObject
+    ) {
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java]: remove the enter line (indent)

File: phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java [hunk 3]
```diff
@@ -511,9 +518,41 @@ private POSTResultsReturn checkAndInsertUserList(List<UserDTO> newUsers) throws
                     insertPreparedStatementAtGroupUsers.executeBatch();
                 }
                 connection.commit(); //Envoi des données dans la BD
-////////////////////
-//ATTENTION, vérifications à re regarder et re vérifier
-//////////////////
+
+                //SILEX:conception
+                // Suggestion : ? create user uri directly from firstname and lastname
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java]: can you be more precise in the comment ? I don't understand exactly the suggestion

File: phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java [hunk 4]
```diff
@@ -511,9 +518,41 @@ private POSTResultsReturn checkAndInsertUserList(List<UserDTO> newUsers) throws
...
+                // Insert URI
+                UriGenerator uriGenerator = new UriGenerator();
+                URINamespaces uriNamespaces = new URINamespaces();
+                final String insertUserURI = "UPDATE \"users\""
+                        + "SET \"uri\" = ? WHERE \"email\" = ?";
+                insertPreparedStatementUserUri = connection.prepareStatement(insertUserURI);
+                for (User u : users) {
+                    if (existInDB(u)) {
+                        // add new URI to user
+                        String userUri = uriGenerator.generateNewInstanceUri(uriNamespaces.getObjectsProperty("cPerson"), null, u.getEmail());
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java]: why did you do another query to insert the uri in the database ? You should just add the uri attribute to the user insertion query.

File: phis2-ws/src/main/java/phis2ws/service/configuration/URINamespaces.java [hunk 5]
```diff
@@ -148,14 +150,17 @@ private void setObjects() {
         OBJECTS.put("cVector", CONTEXTS.get("pVoc2017") + "#Vector");
         OBJECTS.put("cSensingDevice", CONTEXTS.get("pVoc2017") + "#SensingDevice");
         OBJECTS.put("cRestriction", "owl:Restriction");
-        OBJECTS.put("cAgent", CONTEXTS.get("pxFoaf") + "Person");
+        OBJECTS.put("cAgent", CONTEXTS.get("pxFoaf") + "/Agent");
+        OBJECTS.put("cPerson", CONTEXTS.get("pxFoaf") + "/Person");
+        OBJECTS.put("cMotivation", CONTEXTS.get("pxOa") + "#Motivation");
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/configuration/URINamespaces.java]: why is there cMotivation and cAnnotation in the user email add

File: phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java (hunk 2 above)
Comment (niio972) [reply phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java]: Done

File: phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java (hunk 4 above)
Comment (niio972) [reply phis2-ws/src/main/java/phis2ws/service/dao/phis/UserDaoPhisBrapi.java]: Modify

File: phis2-ws/src/main/java/phis2ws/service/configuration/URINamespaces.java (hunk 5 above)
Comment (niio972) [reply phis2-ws/src/main/java/phis2ws/service/configuration/URINamespaces.java]: Removed

**Pull request merged**